watchtower/wtclient/backup_task.go

```diff
@@ -0,0 +1,344 @@
+//  3. Send: Once the task is bound, it will be queued to send to a specific
+//     tower corresponding to the session in which it was bound. The justice
+//     transaction will be assembled by examining the parameters left as a
+//     result of the binding. After the justice transaction is signed, the
+//     necessary components are stripped out and encrypted before being sent to
+//     the tower in a StateUpdate.
+type backupTask struct {
+	chanID       lnwire.ChannelID
+	commitHeight uint64
+	breachInfo   *lnwallet.BreachRetribution
+
+	// state-dependent variables
+
+	toLocalInput  input.Input
+	toRemoteInput input.Input
+	totalAmt      btcutil.Amount
+
+	// session-dependent variables
+
+	blobType      blob.Type
+	sweepAmt      int64
+	rewardAmt     int64
+	rewardAddress []byte
+}
+
+// newBackupTask initializes a new backupTask and populates all state-dependent
+// variables.
+func newBackupTask(chanID *lnwire.ChannelID,
+	breachInfo *lnwallet.BreachRetribution) *backupTask {
+
+	// Parse the non-dust outputs from the breach transaction,
+	// simultaneously computing the total amount contained in the inputs
+	// present. We can't compute the exact output values at this time
+	// since the task has not been assigned to a session, at which point
+	// parameters such as fee rate, number of outputs, and reward rate will
+	// be finalized.
+	var (
+		totalAmt      int64
+		toLocalInput  input.Input
+		toRemoteInput input.Input
+	)
+
+	// If the to-local output is non-dust, add it to the set of candidate
+	// inputs and total input amount.
+	if breachInfo.RemoteOutputSignDesc != nil {
+		toLocalInput = input.NewBaseInput(
+			&breachInfo.RemoteOutpoint,
+			input.CommitmentRevoke,
+			breachInfo.RemoteOutputSignDesc,
+			0,
+		)
+		totalAmt += breachInfo.RemoteOutputSignDesc.Output.Value
+	}
+
+	// If the to-remote output is non-dust, add it to the set of candidate
+	// inputs and total input amount.
+	if breachInfo.LocalOutputSignDesc != nil {
+		toRemoteInput = input.NewBaseInput(
+			&breachInfo.LocalOutpoint,
+			input.CommitmentNoDelay,
+			breachInfo.LocalOutputSignDesc,
+			0,
+		)
+		totalAmt += breachInfo.LocalOutputSignDesc.Output.Value
+	}
+
+	return &backupTask{
+		chanID:        *chanID,
+		commitHeight:  breachInfo.RevokedStateNum,
+		breachInfo:    breachInfo,
+		toLocalInput:  toLocalInput,
+		toRemoteInput: toRemoteInput,
+		totalAmt:      btcutil.Amount(totalAmt),
+	}
+}
+
+// inputs returns all non-dust inputs that we will attempt to spend from.
+//
+// NOTE: Ordering of the inputs is not critical as we sort the transaction with
+// BIP69.
+func (t *backupTask) inputs() map[wire.OutPoint]input.Input {
+	inputs := make(map[wire.OutPoint]input.Input)
+	if t.toLocalInput != nil {
+		inputs[*t.toLocalInput.OutPoint()] = t.toLocalInput
+	}
+	if t.toRemoteInput != nil {
+		inputs[*t.toRemoteInput.OutPoint()] = t.toRemoteInput
+	}
+	return inputs
+}
+
+// bindSession determines if the backupTask is compatible with the passed
```

Comment:
The comment references validation, but I don't see any in the actual method. Perhaps it should be an error if _both_ the local+remote outputs aren't found?

Reply:
This is handled when computing the amounts, it will fail since the totalAmt is 0 and this isn't enough to cover the txn fee